core/shared/src/main/scala/coursier/util/Parse.scala

```diff
@@ -242,6 +262,28 @@ object Parse {
       case _ =>
         Left(s"Malformed dependency: $s")
     }
+
+    depOrError.right.map(dep => (dep, extraDependencyParams))
+  }
+
+  /**
+   * Validates the parsed attributes
+   *
+   * Currently only "classifier" and "url" are allowed. If more are
+   * added, they should be passed in via the second parameter
+   *
+   * @param attrs Attributes parsed
+   * @param validAttrsKeys Valid attribute keys
+   * @return A string if there is an error, otherwise None
+   */
+  private def validateAttributes(attrs: Map[String, String], validAttrsKeys: Set[String]): Option[String] = {
+    val extraAttributes = attrs.keys.toSet.diff(validAttrsKeys)
+
+    if (attrs.size > validAttrsKeys.size || extraAttributes.nonEmpty)
+      Some(s"The only attributes allowed are: ${validAttrsKeys.mkString(", ")}. ${
+        if (extraAttributes.nonEmpty) s"The following are invalid: ${extraAttributes.mkString(", ")}"
```

Comment:
nit: I feel the error can be more descriptive, e.g. for `junit:junit:4.12,abc=123`:
```
The only attributes allowed are: classifier, url. The following attribute(s) are invalid: 'abc' in 'junit:junit:4.12,abc=123'
```